src/components/__tests__/SearchBar.spec.ts

```diff
@@ -0,0 +1,167 @@
+                },
+                createElement("div", { className: "search-container" },
+                    createElement("span", { className: "glyphicon glyphicon-search" }),
+                    createElement("input", { className: "form-control", placeholder: "Search" }),
+                    createElement("button", { className: "btn-transparent" },
+                        createElement("span", { className: "glyphicon glyphicon-remove" })
+                    )
+                )
+            )
+        );
+    });
+
+    describe("input", () => {
+        it("accepts value", () => {
+            const searchBarProps: SearchBarProps = {
+                defaultQuery: "search bar",
+                onTextChangeAction:  () => { return; },
+                placeHolder: "",
+                showSearchBar: true,
+                style: SearchBar.parseStyle("html{}")
+            };
+
+            const searchBarComponent = TestUtils.renderIntoDocument(createElement(SearchBar, searchBarProps));
+            const inputField = TestUtils.findRenderedDOMComponentWithTag(searchBarComponent, "input");
+            TestUtils.Simulate.change(inputField);
+
+            expect((inputField as HTMLInputElement).value).toBe("search bar");
+        });
+
+        it("has input when default query is specified", () => {
+            const searchBarProps: SearchBarProps = {
+                defaultQuery: "default",
+                onTextChangeAction:  jasmine.any(Function) as any,
+                placeHolder: "",
+                showSearchBar: true,
+                style: SearchBar.parseStyle("html{}")
+            };
+            const searchBar = renderSearchBar(searchBarProps);
+
+            expect(searchBar).toBeElement(
+                createElement("div",
+                    {
+                        className: "widget-offline-search",
+                        style: SearchBar.parseStyle("{{...}}")
+                    },
+                    createElement("div", { className: "search-container" },
+                        createElement("span", { className: "glyphicon glyphicon-search" }),
+                        createElement("input", { className: "form-control", placeholder: "", value: "default" }),
+                        createElement("button", { className: "btn-transparent" },
+                            createElement("span", { className: "glyphicon glyphicon-remove" })
+                        )
+                    )
+                )
+            );
+        });
+
+        it("updates when the search changes", () => {
+            const searchBarProps: SearchBarProps = {
+                defaultQuery: "search bar",
+                onTextChangeAction:  () => { return; },
+                placeHolder: "",
+                showSearchBar: true,
+                style: Object.prototype
+            };
+
+            let searchBarComponent = TestUtils.renderIntoDocument(createElement(SearchBar, searchBarProps));
```

Comment:
These untils are short cut for doing shallow render....
They make writing the tests way more easier, fun , quick and clean. 
However i am dropping them just due to organization preferences